include/GraphCoarsen.hpp

```diff
@@ -0,0 +1,208 @@
+                 int max_evects, double spect_tol);
+
+    /** @brief Default Destructor */
+    ~GraphCoarsen() noexcept = default;
+
+    /** @brief Copy Constructor */
+    GraphCoarsen(const GraphCoarsen& other) noexcept;
+
+    /** @brief Move Constructor */
+    GraphCoarsen(GraphCoarsen&& other) noexcept;
+
+    /** @brief Assignment Operator */
+    GraphCoarsen& operator=(GraphCoarsen other) noexcept;
+
+    /** @brief Swap to coarseners */
+    friend void swap(GraphCoarsen& lhs, GraphCoarsen& rhs) noexcept;
+
+    /** @brief Create the coarse mixed matrix
+        @param mgl Fine level mixed matrix
+    */
+    MixedMatrix Coarsen(const MixedMatrix& mgl) const;
+
+    /** @brief Interpolate a coarse vertex vector to the fine level
+        @param coarse_vect vertex vector to interpolate
+        @returns fine_vect interpolated fine level vertex vector
+    */
+    Vector Interpolate(const VectorView& coarse_vect) const;
+
+    /** @brief Interpolate a coarse vertex vector up to the fine level
+        @param coarse_vect vertex vector to interpolate
+        @param fine_vect interpolated fine level vertex vector
+    */
+    void Interpolate(const VectorView& coarse_vect, VectorView fine_vect) const;
+
+    /** @brief Restrict a fine level vertex vector to the coarse level
+        @param fine_vect fine level vertex vector
+        @returns coarse_vect restricted vertex vector
+    */
+    Vector Restrict(const VectorView& fine_vect) const;
+
+    /** @brief Restrict a fine level vertex vector to the coarse level
+        @param fine_vect fine level vertex vector
+        @param coarse_vect restricted vertex vector
+    */
+    void Restrict(const VectorView& fine_vect, VectorView coarse_vect) const;
+
+    /** @brief Interpolate a coarse mixed form vector to the fine level
+        @param coarse_vect mixed form vector to interpolate
+        @returns fine_vect interpolated fine level mixed form vector
+    */
+    BlockVector Interpolate(const BlockVector& coarse_vect) const;
+
+    /** @brief Interpolate a coarse mixed form vector up to the fine level
+        @param coarse_vect mixed form vector to interpolate
+        @param fine_vect interpolated fine level mixed form vector
+    */
+    void Interpolate(const BlockVector& coarse_vect, BlockVector& fine_vect) const;
+
+    /** @brief Restrict a fine level mixed form vector to the coarse level
+        @param fine_vect fine level mixed form vector
+        @returns coarse_vect restricted mixed form vector
+    */
+    BlockVector Restrict(const BlockVector& fine_vect) const;
+
+    /** @brief Restrict a fine level mixed form vector to the coarse level
+        @param fine_vect fine level mixed form vector
+        @param coarse_vect restricted mixed form vector
+    */
+    void Restrict(const BlockVector& fine_vect, BlockVector& coarse_vect) const;
+
+    /** @brief Get the face to coarse dof relationship */
+    const SparseMatrix& GetFaceCDof() const { return face_cdof_; }
```

Comment:
This function may be deprecated.

Reply:
(because the relation is store in `MixedMatrix` now)

Reply:
Correct.